backend/app/services/ai/segment_retrieval.py

```
@@ -0,0 +1,197 @@
+
+        # Fallback to text search if no embeddings
+        return await self._text_based_retrieval(query, base_query, max_segments)
+
+    async def _text_based_retrieval(
+        self, query: str, base_query, max_segments: int
+    ) -> list[tuple[Segment, float]]:
+        """Fallback text-based retrieval using keyword matching."""
+        # Simple keyword-based search - handle Chinese text better
+        query_lower = query.lower()
+
+        # For Chinese text, split by common punctuation and also try to extract key terms
```

Flake8 (7.2.0)

[error] line too long (89 > 79 characters)

(E501)